File: microservices/progress-service(spring)/src/main/java/com/cuba/microservices/progress_service/application/dto/HabitDTO.java
package com.cuba.microservices.progress_service.application.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter

public class HabitDTO {
    private UUID habitId;
    private Integer userId;
    private String name;
    private String unit;
    private BigDecimal targetAmount;
    private LocalDate startDate;
    private LocalDate endDate;
    private RepeatRuleDTO repeatRule;
}

File: client/lib/types/habit/habit.dto.ts
import { RepeatRule } from "./repeat-rule";
import type { Milestone } from "./milestone"; // nếu bạn đã tách Milestone như trước

export interface HabitRequest {
  name: string;
  unit: string;
  targetAmount: number;
  startDate: string;
  endDate?: string;
  categoryId?: string | null;
  tagIds?: string[];
  repeatRule: RepeatRule;
  milestones?: Milestone[]; // đừng để any[] nếu có thể
}
